src/serac/numerics/functional/quadrature_data.hpp

```diff
@@ -44,6 +44,59 @@ struct Nothing {
 struct Empty {
 };
 
+template <typename T>
```

Comment:
it looks like I clobbered the doxygen comments when updating the special storage classes for `serac::Nothing` and `serac::Empty`, so I'll go add them back.

The specializations are there so that all quadrature data types (including `Nothing` and `Empty`) can reuse the same kernel template, but there is a fast path for those special cases, since they don't actually need to store / access memory.